sesame/src/main/scala/org/w3/banana/sesame/io/SesameRDFReader.scala

```diff
@@ -75,3 +75,62 @@ class SesameJSONLDReader(implicit val ops: RDFOps[Sesame]) extends AbstractSesam
   def getParser() = new SesameJSONLDParser
 }
 
+
+
+class SesameRDFLoader(implicit val ops: RDFOps[Sesame]) extends RDFLoader[Sesame, Try] {
+   import org.openrdf.rio.Rio
+   import java.net.URL
+   import java.net.HttpURLConnection
+   import java.net.URLConnection
+
+  def load(url: URL): Try[Sesame#Graph] = Try {
+    val connection = getConnection(url)
+    val parser = getParser(url, connection)
+    val triples = new LinkedList[Statement]
+    val collector = new org.openrdf.rio.helpers.StatementCollector(triples) with CollectorFix {
+      val ops: RDFOps[Sesame] = SesameRDFLoader.this.ops
+    }
+    parser.setRDFHandler(collector)
+    val in: InputStream = connection.getInputStream
+    parser.parse(in, url.toString)
+    new LinkedHashModel(triples)
+  }
+
+  private def getConnection(url: URL): URLConnection = {
+      val connection = url.openConnection()
+      connection match {
+        case connection: HttpURLConnection =>
+        val acceptHeaderTurtlePriority =
+            "text/n3," +
```

Comment:
text/n3 has to be lower at present, because there are very few parsers that actually parse that. In fact the ordering of these should depend on the parsers available. In any case neither Sesame nor Jena I think have real N3 parsers. Sadly.